x-pack/platform/plugins/shared/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/add_inference_flyout/add_inference_flyout_wrapper.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiTitle,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { InferenceForm } from './inference_form';
+
+interface AddInferenceFlyoutWrapperProps {
+  onFlyoutClose: () => void;
+  resendRequest: () => void;
+}
+
+export const AddInferenceFlyoutWrapper: React.FC<AddInferenceFlyoutWrapperProps> = ({
+  onFlyoutClose,
+  resendRequest,
+}) => {
+  const inferenceCreationFlyoutId = useGeneratedHtmlId({
+    prefix: 'addInferenceFlyoutId',
+  });
+
+  return (
+    <EuiFlyout
+      ownFocus
+      onClose={onFlyoutClose}
+      aria-labelledby={inferenceCreationFlyoutId}
+      data-test-subj="create-inference-flyout"
+    >
+      <EuiFlyoutHeader hasBorder data-test-subj="create-inference-flyout-header">
```

Comment:
This code looks like a pretty close copy of https://github.com/elastic/kibana/blob/main/x-pack/solutions/search/plugins/search_inference_endpoints/public/components/add_inference_endpoints/add_inference_flyout_wrapper.tsx - there should be a lot of component reuse between the inference endpoints plugin and index management. Specifically, the flyout and endpoint dropdown list should be reused.